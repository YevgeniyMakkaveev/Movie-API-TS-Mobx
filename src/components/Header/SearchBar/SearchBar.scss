@import '../../../style/color.scss';
.search {
 background-color: transparent;
 border: 2px solid $primary-color;
 border-radius: 50px;
 font-size: 2rem;
 border: 0.1em solid $secondary-color;
 padding: 0.5rem 1rem;
 width: 80vw;
 color: white;
 margin: 1.5em 0em;

 &:focus {
  outline: none;
  background-color: $primary-color;
 }
}