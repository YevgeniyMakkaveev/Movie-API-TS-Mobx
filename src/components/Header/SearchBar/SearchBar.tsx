import React from "react";
import './SearchBar.scss'

interface ISearchBar {
  onInput: (value: React.SetStateAction<string>) => void;
  onSearch: (e: React.FormEvent<HTMLFormElement>) => void;
  search: string;
}

const SearchBar: React.FC<ISearchBar> = ({ search, onSearch, onInput }) => (
  <form onSubmit={(e) => onSearch(e)}>
    <input
      placeholder="Start search and press enter"
      className="search"
      type="search"
      onChange={(e) => onInput(e.target.value)}
      value={search}
    />
  </form>
);

export default SearchBar;
