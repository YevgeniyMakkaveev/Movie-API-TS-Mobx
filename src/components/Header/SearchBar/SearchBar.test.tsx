import React from 'react';
import { render, screen } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import userEvent from '@testing-library/user-event';

import SearchBar from './SearchBar';

describe('it should be used to test snapshots', () => {
  it('should mach the snapshot', () => {
    const app = render(<SearchBar search="" onInput={() => {}} onSearch={() => {}}  />);
    expect(app).toMatchSnapshot(); 
  });
});

describe('Search component works', () => {
  it('should have a placeholder', () => {
    render(<SearchBar search="" onInput={() => {}} onSearch={() => {}} />);
    expect(screen.queryByPlaceholderText('Start search and press enter')).toBeInTheDocument();
  });
  it('should handle search', () => {
    const onChange = jest.fn();
    render(<SearchBar search="" onInput={onChange} onSearch={() => {}} />);
    userEvent.type(screen.getAllByPlaceholderText('Start search and press enter')[0], 'React');
    expect(onChange).toHaveBeenCalledTimes(5);
  });
});
