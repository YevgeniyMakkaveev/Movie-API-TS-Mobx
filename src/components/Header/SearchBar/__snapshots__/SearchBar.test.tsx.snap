// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should be used to test snapshots should mach the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <input
          class="search"
          placeholder="Start search and press enter"
          type="search"
          value=""
        />
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <input
        class="search"
        placeholder="Start search and press enter"
        type="search"
        value=""
      />
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
